### ISSUE TITLE
StartDate dont work

### ISSUE THREAD

**Issue by Thommii** [Sleepw4lker/TameMyCerts#6]
Hello,
I have successfully installed and set up TameMyCerts (Version 1.3.683.747) (test environment).
Roughly everything works, but it doesn't accept the StartDate...

I created the following policy for testing the certificate template:
```
<!--
  This is an example configuration file for computer online certificate requests.
  Ensures that all certificates issued for this template expire on 31st December 2022, 23:59:59 (Europe/Berlin time).
  Works the same way for offline requests, but should include syntax and directory mapping rules there.
-->
<CertificateRequestPolicy xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Subject>
    <SubjectRule>
        <Field>commonName</Field>
        <MaxLength>64</MaxLength>
        <Patterns>
            <Pattern>
                <Expression>^.*$</Expression>
            </Pattern>
        </Patterns>
    </SubjectRule>
    <SubjectRule>
        <Field>organizationName</Field>
        <MaxLength>64</MaxLength>
        <Patterns>
            <Pattern>
                <Expression>^.*$</Expression>
            </Pattern>
        </Patterns>
    </SubjectRule>
    <SubjectRule>
        <Field>organizationalUnitName</Field>
        <MaxLength>64</MaxLength>
        <Patterns>
            <Pattern>
                <Expression>^.*$</Expression>
            </Pattern>
        </Patterns>
    </SubjectRule>
    <SubjectRule>
        <Field>emailAddress</Field>
        <MaxLength>64</MaxLength>
        <Patterns>
            <Pattern>
                <Expression>^.*$</Expression>
            </Pattern>
        </Patterns>
    </SubjectRule>
    <SubjectRule>
        <Field>localityName</Field>
        <MaxLength>64</MaxLength>
        <Patterns>
            <Pattern>
                <Expression>^.*$</Expression>
            </Pattern>
        </Patterns>
    </SubjectRule>
    <SubjectRule>
        <Field>countryName</Field>
        <MaxLength>64</MaxLength>
        <Patterns>
            <Pattern>
                <Expression>^.*$</Expression>
            </Pattern>
        </Patterns>
    </SubjectRule>
  </Subject>
</CertificateRequestPolicy>
```

The following Flags are set:
```
C:\Users\Administrator.FIRMA>certutil -v -getreg Policy\Editflags
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\CertSvc\Configuration\Firma CA01\PolicyModules\TameMyCerts.Policy\EditFlags:

  EditFlags REG_DWORD = 11016e (1114478)
      (EDITF_ENABLEREQUESTEXTENSIONS -- 1)
    EDITF_REQUESTEXTENSIONLIST -- 2
    EDITF_DISABLEEXTENSIONLIST -- 4
    EDITF_ADDOLDKEYUSAGE -- 8
      (EDITF_ADDOLDCERTTYPE -- 10 (16))
    EDITF_ATTRIBUTEENDDATE -- 20 (32)
    EDITF_BASICCONSTRAINTSCRITICAL -- 40 (64)
      (EDITF_BASICCONSTRAINTSCA -- 80 (128))
    EDITF_ENABLEAKIKEYID -- 100 (256)
      (EDITF_ATTRIBUTECA -- 200 (512))
      (EDITF_IGNOREREQUESTERGROUP -- 400 (1024))
      (EDITF_ENABLEAKIISSUERNAME -- 800 (2048))
      (EDITF_ENABLEAKIISSUERSERIAL -- 1000 (4096))
      (EDITF_ENABLEAKICRITICAL -- 2000 (8192))
      (EDITF_SERVERUPGRADED -- 4000 (16384))
      (EDITF_ATTRIBUTEEKU -- 8000 (32768))
    EDITF_ENABLEDEFAULTSMIME -- 10000 (65536)
      (EDITF_EMAILOPTIONAL -- 20000 (131072))
      (EDITF_ATTRIBUTESUBJECTALTNAME2 -- 40000 (262144))
      (EDITF_ENABLELDAPREFERRALS -- 80000 (524288))
    EDITF_ENABLECHASECLIENTDC -- 100000 (1048576)
      (EDITF_AUDITCERTTEMPLATELOAD -- 200000 (2097152))
      (EDITF_DISABLEOLDOSCNUPN -- 400000 (4194304))
      (EDITF_DISABLELDAPPACKAGELIST -- 800000 (8388608))
      (EDITF_ENABLEUPNMAP -- 1000000 (16777216))
      (EDITF_ENABLEOCSPREVNOCHECK -- 2000000 (33554432))
      (EDITF_ENABLERENEWONBEHALFOF -- 4000000 (67108864))
CertUtil: -getreg-Befehl wurde erfolgreich ausgeführt.
```

The following command is used and executed without errors:
![grafik](https://user-images.githubusercontent.com/76774979/205021226-9e50b271-6cb4-4e62-a923-a3206a2357e5.png)

However, the certificate accepts the expiry date/time as intended, but not the start date/time. This will continue to build as usual:
![grafik](https://user-images.githubusercontent.com/76774979/205022720-c53335e2-5769-4c1c-b363-2142f97cd27c.png)

Have I forgotten to set something or why does the StartDate not work?

PS.: Can I issue a failed certificate request despite the error "ERROR_INVALID_TIME" and thus use a StartDate in the past?

**Comment by Sleepw4lker** [comment]
Hello, thanks for reaching out. Unfortunately, I cannot reproduce the issue (works fine in my lab).

Could you please have a look into the attribute table of the requested/issued certificate ("View attributes/extensions")?

Does it appear there?
![2022-12-08 07_33_04](https://user-images.githubusercontent.com/1654791/206375246-9c696232-3d8e-450a-9e0f-416f2e8d88f8.png)

If you like, reach out to me so that we can have a look into this together, e.g. via screen sharing.

**Comment by Thommii** [comment]
Hello,
thank you for your test and reply.

Yes, I have the attributes in the certificate / requirement:
![grafik](https://user-images.githubusercontent.com/76774979/206573120-6004054d-840f-4d0a-af2c-87d4375d0464.png)


We can look at my test setup together. I would contact you shortly via private mail.

**Comment by Sleepw4lker** [comment]
I already found out that Version 1.3 introduced a bug causing StartDate not getting applied when there is no policy file present for a given certificate template. The desired behavior should be that StartDate always gets applied (whether a policy exists or not), given that the EDITF_ATTRIBUTEENDDATE flag is present and the given StartDate is valid (within the boundaries of the original certificate validity).

**Comment by Sleepw4lker** [comment]
Fix has been implemented and will be available soon with v1.4 release.
